frontend/src/app/components/MainInstallButton.js

```diff
@@ -88,16 +102,22 @@ export default class MainInstallButton extends React.Component {
     if (isFirefox && isMobile) {
       return (
         <div>
-          <span data-l10n-id="landingRequiresDesktop">Test Pilot requires Firefox for Desktop on Windows, Mac or Linux</span>
+          <Localized id="landingRequiresDesktop">
+            <span>Test Pilot requires Firefox for Desktop on Windows, Mac or Linux</span>
+          </Localized>
         </div>
       );
     }
     return (
       <div>
           {!isFirefox ? (
-              <span data-l10n-id="landingDownloadFirefoxDesc" className="parens">(Test Pilot is available for Firefox on Windows, OS X and Linux)</span>
+              <Localized id="landingDownloadFirefoxDesc">
+                <span className="parens">(Test Pilot is available for Firefox on Windows, OS X and Linux)</span>
+              </Localized>
             ) : (
-              <span className="parens" data-l10n-id="landingUpgradeDesc2" data-l10n-args={JSON.stringify({ version: config.minFirefoxVersion })}>Test Pilot requires Firefox { config.minFirefoxVersion } or higher.</span>
+              <Localized id="landingUpgradeDesc2" $version={config.minFirefoxVersion}>
+                <span className="parens">Test Pilot requires Firefox { config.minFirefoxVersion } or higher.</span>
```

Comment:
There are two ways to approach what should go between into the `<span>` here and I'd like to learn your take on which one is more desirable.

The first one is what you did here: put the valid JSX into the `<span>`.  It can serve as a fallback for when the `landingUpgradeDesc2` translation is missing. The drawback is that React will create the virtual DOM nodes for the content which `Localized` will then remove in favor of the translated string.

The other approach is to use valid FTL syntax as `span`'s content:

```
<Localized id="landingUpgradeDesc2" $version={config.minFirefoxVersion}>
  <span className="parens">
    {'Test Pilot requires Firefox { $version } or higher.'}
  </span>
</Localized>
```

This has the benefit of being exactly the string that needs to be translated which in the future could be used for extracting English strings into an FTL file. On the other hand, it doesn't work as well as the first approach when it comes to fallback.  it's just a static string and in particular, `$version` wouldn't get replaced if `landingUpgradeDesc2` is missing from all languages (English included).